pkg/build/api/validation/validation.go

```diff
@@ -682,8 +682,8 @@ func ValidateStrategyEnv(vars []kapi.EnvVar, fldPath *field.Path) field.ErrorLis
 		} else if len(kvalidation.IsCIdentifier(ev.Name)) > 0 {
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), ev.Name, cIdentifierErrorMsg))
 		}
-		if ev.ValueFrom != nil {
-			allErrs = append(allErrs, field.Invalid(idxPath.Child("valueFrom"), ev.ValueFrom, "valueFrom is not supported in build strategy environment variables"))
+		if ev.ValueFrom != nil && ev.ValueFrom.ResourceFieldRef != nil {
```

Comment:
ValidateStrategyEnv is called from ValidateBuildDefaultsConfig - may want to continue to prevent BuildDefaults from using ValueFrom - @bparees?